src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -288,9 +290,29 @@ def delete_arc_agents(release_namespace, kube_config, kube_context, configuratio
     ensure_namespace_cleanup(configuration)
 
 
+def get_container_registry(container_registry_repository):
+    return container_registry_repository.split('/')[0]
+
+
+def helm_login(container_registry_repository, container_registry_username, container_registry_password):
+    if container_registry_username:
+        container_registry = get_container_registry(container_registry_repository)
+        cmd_helm_login = ["helm", "registry", "login", container_registry, "--username", container_registry_username, "--password", container_registry_password]
+        response_helm_install = Popen(cmd_helm_login, stdout=PIPE, stderr=PIPE)
+        _, error_helm_install = response_helm_install.communicate()
+        if response_helm_install.returncode != 0:
+            if ('forbidden' in error_helm_install.decode("ascii") or 'timed out waiting for the condition' in error_helm_install.decode("ascii")):
```

Comment:
I think we can generally set non success as user fault as the credentials are provided by the user. This scoped check should not be reqd. Are you aware of any failure scenario in this step which would not be user fault?

Reply:
ack